src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/ClientManager.java

```diff
@@ -0,0 +1,122 @@
+package com.snowflake.kafka.connector.internal.ingestsdk;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.internal.LoggerHandler;
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+public class ClientManager {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
+
+  /** This is a wrapper to help manage the streaming ingest client */
+  protected ClientManager() {
+    LOGGER = new LoggerHandler(this.getClass().getName());
+    this.streamingIngestClientCount = 0;
+  }
+
+  // ONLY FOR TESTING - use this to inject client
+  @VisibleForTesting
+  public ClientManager(SnowflakeStreamingIngestClient client) {
```

Comment:
You could always have a static Client Manager factory that you could use in lieu of this.

Reply:
Take a look at this thread: https://stackoverflow.com/questions/929021/what-are-static-factory-methods

Reply:
It would work, but i'm not sure i see the benefits over what we use currently. I think it would provide the same benefits for a little more complex code? Happy to discuss further though